test/vanilla/Expected/AcceptanceTests/ExtensibleEnums/Models/DaysOfWeekExtensibleEnum.cs

```diff
@@ -0,0 +1,115 @@
+// <auto-generated>
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+//
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+// </auto-generated>
+
+namespace Fixtures.AcceptanceTestsExtensibleEnums.Models
+{
+    using Newtonsoft.Json;
+
+    /// <summary>
+    /// Defines values for DaysOfWeekExtensibleEnum.
+    /// </summary>
+    /// <summary>
+    /// Determine base value for a given allowed value if exists, else return
+    /// the value itself
+    /// </summary>
+    [JsonConverter(typeof(DaysOfWeekExtensibleEnumConverter))]
+    public struct DaysOfWeekExtensibleEnum : System.IEquatable<DaysOfWeekExtensibleEnum>
+    {
+        private DaysOfWeekExtensibleEnum(string underlyingValue)
+        {
+            UnderlyingValue=underlyingValue;
+        }
+
+        public static readonly DaysOfWeekExtensibleEnum Monday = "Monday";
```

Comment:
I have a feature request that will help us move the Azure Search SDK to use this new feature. Please generate a nested class to contain the string constants, and then refer to them here. 

For example, see the `AsString` class in this file: https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Customizations/Indexes/Models/AnalyzerName.cs

The reason we need this is that sometimes we need to refer to one of the extensible enum values from the parameter to an attribute, and such parameters must be compile-time constants.

Reply:
Seems doable and looks like something we can implement incrementally, will open an issue describing this feature requirement.